server/broadcast.go

```diff
@@ -811,6 +811,11 @@ func processSegment(cxn *rtmpConnection, seg *stream.HLSSegment) ([]string, erro
 			glog.Warningf("Not retrying current segment nonce=%d seqNo=%d due to non-retryable error err=%v", nonce, seg.SeqNo, err)
 			break
 		}
+		if ctxErr := ctx.Err(); ctxErr != nil {
+			err = ctxErr
+			glog.Warningf("Not retrying current segment nonce=%d seqNo=%d due to error err=%v", nonce, seg.SeqNo, err)
```

Comment:
**nit:** just for debugging swiftness, we can be a little more explicit here. WDYT?
```suggestion
			glog.Warningf("Not retrying current segment nonce=%d seqNo=%d due to context cancellation err=%v", nonce, seg.SeqNo, err)
```

Reply:
Good idea, applied.